components/file_attachment/index.ts

```diff
@@ -7,16 +7,20 @@ import {getConfig} from 'mattermost-redux/selectors/entities/general';
 
 import {GlobalState} from 'types/store';
 
+import {FileDropdownPluginComponent} from 'types/store/plugins';
 import {canDownloadFiles} from 'utils/file_utils';
 
 import FileAttachment from './file_attachment';
 
 function mapStateToProps(state: GlobalState) {
     const config = getConfig(state);
+    const pluginMenuItems = (state.plugins.components.FilesDropdown || []) as unknown as FileDropdownPluginComponent[];
```

Comment:
If a user has no FilesDropdown plugins, is `state.plugins.components.FilesDropdown` an empty array or null/undefined? If it's null, then this'll cause FileAttachment components to re-render constantly when there's no plugins, so it'll need to be memoized.

Reply:
@hmhealey 
`state.plugins.components.FilesDropdown` comes as `undefined` if no relevant plugin is installed and empty array `[]` if a plugin is installed and then removed.

I've added the memoization for this selector. Please check.